src/T2DMap.cpp

```diff
@@ -198,28 +198,49 @@ void T2DMap::slot_switchArea(const QString& newAreaName)
     }
 
     QMapIterator<int, QString> it(mpMap->mpRoomDB->getAreaNamesMap());
+    TEvent areaViewedChangedEvent{};
     while (it.hasNext()) {
         it.next();
         int areaID = it.key();
-
         auto areaName = it.value();
         TArea* area = mpMap->mpRoomDB->getArea(areaID);
         if (area && newAreaName == areaName) {
-            mAreaID = areaID;
+            if (mAreaID != areaID) {
+                // We are changing the viewed area - so change the zoom to the
+                // stored zoom value - this way though will reset it to the
+                // default if the new area has not previously had a zoom set
+                // from the Lua API and some parties have suggested that in
+                // this case we should just use the zoom value that we were
+                // using in the area we are switching from:
+                xyzoom = mpMap->mpRoomDB->get2DMapZoom(areaID);
+                areaViewedChangedEvent.mArgumentList.append(qsl("mapAreaViewedChangeEvent"));
```

Comment:
I know we went off track in keeping consistency for Mudlet's event names - suggest to bring it back:

```suggestion
                areaViewedChangedEvent.mArgumentList.append(qsl("sysMapAreaChanged"));
```

Reply:
:thinking: 
Of the **64** events Mudlet currently raises, **17** end with `"Event"` (and **7** with `"Error"`), **2** start with `"map"`, **59** start with `"sys"`... so I guess this makes sense and I will do this.

OTOH though, ***I*** do like the "Event" suffix as it gives a clear indication what the string represents when a handler function gets it as the first argument!

Reply:
Done in b307d19ed1549c9c5290e05b6210bf4f0fd54088 (I forgot it when doing the others...)